### ISSUE TITLE
CO2 emissions of electricity production chart does not match its description

### ISSUE THREAD

**Issue by mabijkerk** [quintel/etmodel#3995]
**Problem:** the chart `CO2 emissions of electricity production` contains emissions that arise from electricity production for the products natural gas, coal, lignite, oil and waste.

<img width="500" alt="Screenshot 2022-08-12 at 14 34 40" src="https://user-images.githubusercontent.com/67338510/184354951-659596b6-56bf-47d6-81aa-ed58ec65e62e.png">

The description of the chart states the following:
> This chart only shows the **direct emissions** that arise from domestic electricity production. Indirect emissions from electricity production, for example emissions from hydrogen power plants or from imported electricity, are therefore left out of the chart.

However, the underlying queries, like [coal_in_co2_of_electricity_production](https://github.com/quintel/etsource/blob/master/gqueries/output_elements/output_series/vertical_stacked_bar_162_co2_of_electricity_production/coal_in_co2_of_electricity_production.gql), use the primary co2 emission recursive factor. This means that they don't actually show direct emissions.

**Solution:** either new queries have to be written for this chart that actually calculate the direct emissions, or we should change the description to say that it calculates primary emissions. If we update the description I think we should actually show the primary emissions of **all** power plant types, so also hydrogen, imports etc.

@marliekeverweij do you agree that we should change the chart and if so, what is your preferred solution?

**Comment by marliekeverweij** [comment]
Interesting and good issue! However, I do not have a strong opinion on this. For Northern-Ireland we are using this chart, so maybe @DorinevanderVlies & @MartLubben have input?

@quintel/core if other people have an opinion

**Comment by AlexanderWirtz** [comment]
I agree with @mabijkerk let's include it all! Except if for local energy transition purposes there is a reason for the limited scope

**Comment by DorinevanderVlies** [comment]
> Solution: either new queries have to be written for this chart that actually calculate the direct emissions, or we should change the description to say that it calculates primary emissions. If we update the description I think we should actually show the primary emissions of all power plant types, so also hydrogen, imports etc.

I am not certain about which definition is more common in politics and policy, direct or primary. For EirGrid the direct emissions are relevant. Some other users might prefer primary emissions and perhaps primary emissions is more in line with other charts and calculations in the ETM.

Furthermore we should take care that our charts remain MECE.  Is there a risk of double counting if we include indirect CO2 emissions of for example hydrogen here as well as for hydrogen?

So, in short, I agree that this chart needs to change so it is in line with its description. But, I am no sure yet how we should change it.

Shall we have a live discussion about this?


The current description of the chart is closer to

**Comment by mabijkerk** [comment]
@DorinevanderVlies was there a conclusion about what to do with this chart?

**Comment by mabijkerk** [comment]
I had a discussion with @DorinevanderVlies and we arrived at the following conclusions:

**Summary of issue**

- The chart only shows series that are relevant for direct emissions (so no hydrogen or imported electricity)
- The chart queries the entire electricity output of producers (so not only electricity produced for domestic demand)
- This is clearly stated in the chart description
- The only problem is that it uses the primary emission method to query the emissions on the electricity producer nodes

**Conclusions**
- The chart is useful the way it is
- Solution for this specific issue would be to write a method for ETEngine that allows us to query direct emissions on a node
- The difference with the current method is expected to be small, since not much "happens" in the graph further to the "right" (moving closer towards primary demand nodes) of the electricity producer nodes
- **This is therefore not an urgent issue**
- An enhancement would be to show multiple CO2 accounting methods in this chart
- A more structural enhancement would be to facilitate multiple CO2 accounting methods

**To do**

- [ ] Add a method for querying direct emissions on a node
- [ ] Update the queries to use that method

Structural enhancement should be picked up in a broader project.
